src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -318,8 +318,7 @@ public SNIHandle CreateConnectionHandle(object callbackObject, string fullServer
                     sniHandle = CreateTcpHandle(serverNameWithOutProtocol, timerExpire, callbackObject, parallel, ref spnBuffer, isIntegratedSecurity);
                 }
                 // when np protocol is specified
-                else if ((serverNameWithOutProtocol = GetServerNameWithOutProtocol(fullServerName, TdsEnums.NP + ":\\\\")) != null ||
-                         (serverNameWithOutProtocol = GetServerNameWithOutProtocol(fullServerName, "\\\\")) != null)
+                else if ((serverNameWithOutProtocol = GetServerNameWithOutProtocol(fullServerName, TdsEnums.NP + ":")) != null)
```

Comment:
The GetServerNameWithOutProtocol function trims off the protocol part, so these calls were also cutting off the first backslashes in the pipe paths. The second call in the if statement is also unreachable, since strings with no protocol specified are handled in another case earlier in this function.